pallets/liquidity-pools/src/message.rs

```diff
@@ -740,18 +740,18 @@ mod tests {
 	}
 
 	#[test]
-	fn transfer_tranche_tokens_to_moonbeam() {
-		let domain_address = DomainAddress::EVM(1284, default_address_20());
+	fn transfer_tranche_tokens_to_chain() {
+		let domain_address = DomainAddress::Evm(1284, default_address_20());
 
 		test_encode_decode_identity(
 			Message::TransferTrancheTokens {
 				pool_id: 1,
 				tranche_id: default_tranche_id(),
 				domain: domain_address.domain().into(),
-				receiver: domain_address.address(),
+				receiver: domain_address.as_local(),
```

Comment:
I am skeptical about the changed serialization here. Probably should be
```suggestion
				receiver: domain_address.as_eth(),
```

Reply:
`address()` returned a `[u8; 32]`, the `receiver` has type `[u8; 32]` but `as_eth()` returns an [u8;20]. 

Basically we use `address` as 32-bytes in the message for compatibility reasons, but later the 12 remaining bytes are discarded by solidity, see this thread: https://kflabs.slack.com/archives/C06FRHCRQL9/p1723627567780339

Reply:
Thanks for breaking this down. Seems like the message serialization is now correct and was poorly mocked beforehand, i.e. by using the 20 byte account address which in reality, should never occur.

Reply:
Yeah, previously it was a chaos of different versions of `[u8; 20] -> [u8; 32]`  conversions